### ISSUE TITLE
[Bug]: Checkbox fails accessibility requirement & Update to inverted colors

### ISSUE THREAD

**Issue by redongreen** [microsoft/fluentui#24014]
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/theme-color--page
```

### Reproduction

https://master--628d031b55e942004ac95df1.chromatic.com/?path=/docs/theme-color--page

### Context
The checkbox component currently fails accessibility requirements. This issue was introduced when we updated our Brand Communicate Blue ramp, where we normalized our default color ramp to be less saturated. Daisy created a [proposal to update our controls to use inverted colors](https://www.figma.com/file/edE0nBTGtp5tVToWGNdkNV/Brand-Color?node-id=714%3A258368). This proposal aligns with our [guidance on brand colors](https://www.figma.com/file/xShPc11RUrzGWqZWfR4kFp/Ian-Guisard-Scratchpad?node-id=931%3A198658).

The goal is to;

1. Address accessibility issues with our checkbox component
2. Create coherence between our implementation and guidelines
3. Ensure that we are applying our semantic tokens consistently

**Examples**
(A) The checkbox component fails accessibility
<img width="656" alt="image" src="https://user-images.githubusercontent.com/20573265/180107901-8f665e1e-165f-4568-85ed-3c41df14cc82.png">

(B) Inverted colors should be inverted from light to dark, and dark to light. Currently, `Neutral Background Inverted` is a combination of dark in light and dark in dark theme. `Neutral Foreground Inverted` is light in light theme and light in dark theme.
![image](https://user-images.githubusercontent.com/20573265/180108275-659f1f23-5f98-454d-b250-6460d6c30240.png)

The following design tokens need to be updated;
___
### [Change] Neutral Foreground Inverted value

**From**
```
[Light] Neutral Foreground Inverted = White
[Light] Neutral Foreground Inverted Hover = White
[Light] Neutral Foreground Inverted Pressed = White
[Light] Neutral Foreground Inverted Selected = White

[Dark] Neutral Foreground Inverted = White
[Dark] Neutral Foreground Inverted Hover = White
[Dark] Neutral Foreground Inverted Pressed = White
[Dark] Neutral Foreground Inverted Selected = White
```

**To**

- [x] [Light] Neutral Foreground Inverted = White
- [x] [Light] Neutral Foreground Inverted Hover = White
- [x] [Light] Neutral Foreground Inverted Pressed = White
- [x] [Light] Neutral Foreground Inverted Selected = White
- [x] [Dark] Neutral Foreground Inverted = Grey 14
- [x] [Dark] Neutral Foreground Inverted Hover = Grey 14
- [x] [Dark] Neutral Foreground Inverted Pressed = Grey 14
- [x] [Dark] Neutral Foreground Inverted Selected = Grey 14
- [x] [High Contrast] Neutral Foreground Inverted (rest, hover, pressed, selected) = HighlightText (#000000)
- [x] [Teams Dark] Neutral Foreground Inverted (rest, hover, pressed, selected) = Grey 14

Impacted components

- [x] Checkbox (impacted but no needs needed)
- [x] ~Tooltip FG (change Tooltip FG to Neutral Foreground Static Inverted)~
    - Change is for token below
- [x] ~Spinner FG (change Subtle FG to Neutral Foreground Static Inverted)~
    - Change is for token below

___
### [Add] Neutral Foreground Static Inverted

 **New design token:**

- [x] [Light] Neutral Foreground Static Inverted = White
- [x] [Dark] Neutral Foreground Static Inverted = White
- [x] [High Contrast] Neutral Foreground Static Inverted = hcCanvasText (#FFFFFF)
- [x] [Teams Dark] Neutral Foreground Static Inverted = White

Description: Inverted static foreground

Impacted components

- [x] Tooltip FG (change Tooltip FG to Neutral Foreground Static Inverted)
- [x] Spinner FG (change Subtle FG to Neutral Foreground Static Inverted)

___
### [Change] Neutral Background Inverted value

**From**
```
[Light] Neutral Background Inverted = Grey 38
[Dark] Neutral Background Inverted = Grey 24
```

**To**

- [x] [Light] Neutral Background Inverted = Grey 16
- [x] [Dark] Neutral BackgroundInverted = White
- [x] [High Contrast] Neutral BackgroundInverted = hcCanvas (#000000)
- [x] [Teams Dark] Neutral BackgroundInverted = White

Impacted components

- [x] Tooltip BG (change Tooltip BG to Neutral Background Static)
- [x] Popover BG (change Popover BG to Neutral Background Static)

___
### [Add] Neutral Background Static

- [x] [Light] Neutral Background Static = Grey 20
- [x] [Dark] Neutral Background Static = Grey 24
- [x] [High Contrast] Neutral Background Static = hcCanvas (#000000)
- [x] [Teams Dark] Neutral Background Static = Grey 24

Description: Static background surface

Impacted components

- [x] Tooltip BG (change Tooltip BG to Neutral Background Static)
- [x] Popover BG (change Popover BG to Neutral Background Static)

___
### Components not directly impacted but must be updated 
**Checkbox (checked)**

- [x] Change the checkbox “checkmark” from `Neutral Foreground On Brand` to `Neutral Foreground Inverted`

**Switch (checked)**

- [x] Change the switch knob from `Neutral Background 1` to `Neutral Foreground Inverted`

### Requested priority

High

### Products/sites affected

Fluent 2 WEB UI

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

**Comment by redongreen** [comment]
@spmonahan 

I updated my comment with the following correct;

Previously I said:
[Light] Neutral Background Inverted = **Grey 20** 

The actual value should be:
[Light] Neutral Background Inverted = **Grey 16**

**Comment by spmonahan** [comment]
The "brand" appearances for Popover looks like it needs an update text color. There's [a comment about it in source](https://github.com/microsoft/fluentui/blob/master/packages/react-components/react-popover/src/components/PopoverSurface/usePopoverSurfaceStyles.ts#L37-L38) and it has some clear contrast issues in most themes.

One example:
![popover_brand_appearance_theme_web_dark](https://user-images.githubusercontent.com/93940821/180573293-05dc60a9-e04e-4f58-a43e-03814297c165.png)

**Comment by redongreen** [comment]
For brand popovers use the following [(based on our guidance)](https://www.figma.com/file/GvIcCw0tWaJVDSWD4f1OIW/Fluent-2-Web?node-id=24429%3A135089);
Background: Brand Background
Foreground: Brand Foreground On Color

<img width="481" alt="image" src="https://user-images.githubusercontent.com/20573265/180577946-77072f40-3bee-4ffe-83ad-8fd8a2e00a22.png">

**Comment by spmonahan** [comment]
Note: the token value changes also affected `Badge` and `Avatar`. The linked PR has changes for those components as well.

**Comment by spmonahan** [comment]
Related: #24161
